### ISSUE TITLE
The Options modal closes immediately when checking an option and blocks are in navigation mode

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#24190]
**Describe the bug**

Tested in WordPress 5.5 Beta 3:

Checking the option "Contain text cursor inside active block" makes the Options modal dialog close immediately but only when the block editor is in "navigation mode".

Expected: the Options modal dialog to stay open.

**To reproduce**
- click on a block: the editor is now in "Edit mode"
- press the Escape key: the editor is now in "Navigation mode"
- open the editor More menu and click on "Options" 
- the Options modal dialog opens
- click on "Contain text cursor inside active block"
- the modal dialog closes unexpectedly
- click on a block and keep the editor in "Edit mode"
- reopen the modal, click on "Contain text cursor inside active block"
- this time, the modal stays open, as expected
- repeat the steps above clicking on other options in the modal dialog
- regardless of navigation mode / edit mode, it seems to happen only for the "Contain text cursor inside active block" option

[Edit]
Correction: it seems to happen also when clicking "Enable the Most Used Blocks category in the block library".

**Comment by afercia** [comment]
I'd like to note this is not a regression but it's a _new feature in WordPress 5.5_ that doesn't meet accessibility (and usability) needs and ideally should be fixed before the 5.5 release.
